lib/jellyfish/component/rtsp.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Jellyfish.Component.RTSP do
+  @moduledoc """
+  Module representing the RTSP component.
+  """
+
+  @behaviour Jellyfish.Endpoint.Config
+
+  alias Membrane.RTC.Engine.Endpoint.RTSP
+
+  alias JellyfishWeb.ApiSpec
+
+  @impl true
+  def config(%{engine_pid: engine} = options) do
+    options = Map.drop(options, [:engine_pid, :room_id])
+
+    with {:ok, valid_opts} <- OpenApiSpex.cast_value(options, ApiSpec.Component.RTSP.schema()) do
```

Comment:
[The docs](https://hexdocs.pm/open_api_spex/3.16.1/readme.html#validating-and-casting-params) describe a way to enable global validation of requests, which can be done by plugging `OpenApiSpex.Plug.CastAndValidate` in the component controller.

I tried to use that option but couldn't get it to work. Will look more into it.

Reply:
After consideration, we've decided to stick with the current approach when it comes to validation.